lib/widgets/sign_up_widget.dart

```diff
@@ -18,25 +23,37 @@ class SignUp extends StatefulWidget {
 class _SignUpState extends State<SignUp> {
   bool isLoading = false;
 
-  AuthMethods authMethods = new AuthMethods();
+  AuthMethods authMethods = AuthMethods();
+  DatabaseMethods databaseMethods = DatabaseMethods();
 
   final formKey = GlobalKey<FormState>();
-  TextEditingController userNameTextEditingController =
-      new TextEditingController();
-  TextEditingController emailTextEditingController =
-      new TextEditingController();
-  TextEditingController passwordTextEditingController =
-      new TextEditingController();
 
-  SignMeUp() {
+  TextEditingController userNameTextEditingController = TextEditingController();
+  TextEditingController emailTextEditingController = TextEditingController();
+  TextEditingController passwordTextEditingController = TextEditingController();
+
+  signUp() {
     if (formKey.currentState.validate()) {
+      Map<String, String> userInfoMap = {
```

Comment:
try to avoid injecting business logic into UI. For this moment it's minor but read about application's architecture design patterns